### PR TITLE
Added pip, pipenv, yarn support to Frogbot PR fix

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.

---

### CODE REVIEW

File: commands/createfixpullrequests.go [hunk 1]
```diff
@@ -16,6 +19,12 @@ import (
 	"github.com/jfrog/jfrog-client-go/xray/services"
 )
 
+// case insensitive
```
Comment (eyalbe4) [inline commands/createfixpullrequests.go:22]: case --> Case

Let's elaborate more and describe this variable.

File: commands/createfixpullrequests.go [hunk 2]
```diff
@@ -212,48 +226,95 @@ func (cfp *CreateFixPullRequestsCmd) fixSinglePackageAndCreatePR(impactedPackage
...
+		commandArgs := []string{"install"}
+		err = fixPackageVersionGeneric(commandArgs, coreutils.Pipenv, impactedPackage, fixVersion, "==")
+	default:
+		return fmt.Errorf("package type: %s is currently not supported", string(packageType))
+	}
+
+	return err
+}
+
+// The majority of package managers already support upgrading specific package versions and update the dependency files automatically
```
Comment (eyalbe4) [inline commands/createfixpullrequests.go:255]: automatically --> automatically.
(missing period at the end of the sentence).

File: commands/createfixpullrequests.go [hunk 3]
```diff
@@ -212,48 +226,95 @@ func (cfp *CreateFixPullRequestsCmd) fixSinglePackageAndCreatePR(impactedPackage
...
+// In other cases, we had to handle the upgrade process
```
Comment (eyalbe4) [inline commands/createfixpullrequests.go:256]: Let's also describe the function's arguments.

File: commands/createfixpullrequests_test.go [hunk 4]
```diff
@@ -1,12 +1,100 @@
...
+	}
+}
+
+func getPipFixPackageVersionFunc() FixPipPackagesTestFunc {
+	return func(test packageFixTest) error {
+		return fixPackageVersionPip(test.impactedPackaged, test.fixVersion, test.packageDescriptor)
+	}
+}
+
+func Test_fixPackageVersion(t *testing.T) {
```
Comment (eyalbe4) [inline commands/createfixpullrequests_test.go:81]: Test_fixPackageVersion --> fixPackageVersionTest

Let's use the same naming structure for all the tests functions throughout the code (not only in this file).

File: commands/createfixpullrequests_test.go [hunk 5]
```diff
@@ -1,12 +1,100 @@
...
+	testdataDir, err := filepath.Abs(filepath.Join("testdata/projects"))
+	assert.NoError(t, err)
+	for _, test := range packageFixTests {
+		// Create temp technology project
+		projectPath := filepath.Join(testdataDir, test.technology)
+		tmpProjectPath, cleanup := testdatautils.CreateTestProject(t, projectPath)
+		defer cleanup()
+		assert.NoError(t, os.Chdir(tmpProjectPath))
```
Comment (eyalbe4) [inline commands/createfixpullrequests_test.go:89]: We have a utility function that changes the dir, and also allows changing the dir back at the end.
You can sonsult with @yahavi about this.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's also update the README to include all the relevant information.

**Comment by omerzi** [comment]
tests passed on my branch (added pipenv installation to test.yml) - https://github.com/omerzi/frogbot/actions/runs/2904024807

**Pull request merged**